In [1]:
import os
import random

used = "used.txt"
if not os.path.exists(used):
    open(used, "w+").close()

with open(used, "r+", encoding="utf8") as f:
    used_problems = f.readlines()
    used_problems = [u.strip("\n") for u in used_problems]
    print("Used:", used_problems)

    r = [i for i in range(1, 719)]
    r = [i for i in r if str(i) not in used_problems]
    random.shuffle(r)
    now = None
    if len(r) == 0:
        print("All done!")
        exit(0)
    elif len(r) == 1:
        now = r[0]
    else:
        now = random.choice(r)
    print("Now:", now)
    f.write(str(now) + "\n")

Used: ['361', '474', '297', '340', '698', '289', '489', '627', '405', '35', '666', '33', '356', '182', '671', '149', '425', '31']
Now: 451


## Sort Characters By Frequency
**Date**: 2018-2-7

**Task**: [451](https://leetcode.com/problems/sort-characters-by-frequency/description/)

**Detail**:

 　Given a string, sort it in decreasing order based on the frequency of characters.

**Example 1:**

```bash
Input:
"tree"

Output:
"eert"

Explanation:
'e' appears twice while 'r' and 't' both appear once.
So 'e' must appear before both 'r' and 't'. Therefore "eetr" is also a valid answer.
```

**Example 2:**

```bash
Input:
"cccaaa"

Output:
"cccaaa"

Explanation:
Both 'c' and 'a' appear three times, so "aaaccc" is also a valid answer.
Note that "cacaca" is incorrect, as the same characters must be together.
```

**Example 3:**

```bash
Input:
"Aabb"

Output:
"bbAa"

Explanation:
"bbaA" is also a valid answer, but "Aabb" is incorrect.
Note that 'A' and 'a' are treated as two different characters.
```